src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/global_navitagtion/NavigationBar.java

```
@@ -18,77 +17,119 @@
-        for (int i = 0; i < suggestionsList.size(); i++) {
-            WebElement currentSuggestion = suggestionsList.get(i);
-            searchInput.sendKeys(Keys.ARROW_DOWN);
-            if (currentSuggestion.getText().contains(suggestion)) {
-                searchInput.sendKeys(Keys.ENTER);
-                PageObjectLogging.log("ArrowDownToSuggestion", 
-                        "arrowed down to desired suggestion" +suggestion+"and clicked enter",
-                        true);
-                return new ArticlePageObject(driver);
-            }
-            else {
-                searchInput.sendKeys(Keys.ARROW_DOWN);
-            }
-        }
-        PageObjectLogging.log("ArrowDownToSuggestion", "didn't find suggestion: "+suggestion, true);
-        return null;
-    }
+	public void triggerSuggestions(String query) {
+		waitForElementByElement(searchInput);
+		searchInput.clear();
+		waitForElementClickableByElement(searchInput);
+	    searchInput.click();
+	    waitForElementByBy(jqueryAutocompleteBy);
+	    sendKeysHumanSpeed(searchInput, query);
+	    waitForElementByCss(suggestionCss);
+	    waitForElementByElement(suggestionsList.get(0));		
+	}
+	
+	public void verifySuggestions(String suggestion) {
+		waitForElementByElement(suggestionsList.get(0));
+		String allSuggestions = "";
+		for (int i = 0; i < suggestionsList.size(); i++) {
+			allSuggestions += suggestionsList.get(i).getAttribute("title");
+		}
+		Assertion.assertStringContains(suggestion, allSuggestions);
+	}
 
-    public IntraWikiSearchPageObject searchFor(String query) {
-        searchInput.sendKeys(query);
-        searchSubmit.click();
-        PageObjectLogging.log("searchFor", "searching for query: " + query, true, driver);
-        return new IntraWikiSearchPageObject(driver);
-    }
+	/**
+    * Arrow down through suggestions, and click enter on the desired one
+	*/
+	public ArticlePageObject ArrowDownAndEnterSuggestion(String suggestion) {
```

use assertion before null instead of pageobjectlogging on fail and success